### PR TITLE
insertion

### PR DESCRIPTION
**Pull request by PallaviT2001**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        }
+        if (temp != NULL) {
+            newnode->next = temp->next;
+            temp->next = newnode;
+        } else {
+            printf("Position not found\n");
+            free(newnode);
+        }
+    }
+    (*count)++;
```
Comment (Maruthi31CF) [inline main.c:127]: Why *count ?

File: main.c [hunk 2]
```diff
@@ -0,0 +1,212 @@
+#include <stdio.h>
+#include <stdlib.h>
```
Comment (Maruthi31CF) [inline main.c:2]: Why is it needed?

File: main.c (hunk 1 above)
Comment (PallaviT2001) [reply main.c:127]: *count is a memory address of the variable count, by using this, function access to the actual memory location of count and directly modify the count value.

File: main.c (hunk 2 above)
Comment (PallaviT2001) [reply main.c:2]: #include <stdio.h> header file is required for standard input/output operations. it provides the declarations for functions like printf(), scanf()
#include <stdlib.h> This header file is required for memory allocation and deallocation ,this header file is necessary in programs that work with data structures like Array, linkedlist etc

**Pull request merged**